src/elements/toniq-dropdown/toniq-dropdown.element.ts

```diff
@@ -0,0 +1,188 @@
+import {assign, css, defineElementEvent, html} from 'element-vir';
+import {ReactiveController, ReactiveControllerHost} from 'lit';
+import {ChevronDown24Icon} from '../../icons';
+import {applyBackgroundAndForeground, toniqColors} from '../../styles/colors';
+import {defineToniqElement} from '../define-toniq-element';
+import {ToniqIcon} from '../toniq-icon/toniq-icon.element';
+
+export interface ToniqDropdownSelect {
+    value: string | number | undefined;
+    label: string | undefined;
+}
+
+export class ToniqDropdownController implements ReactiveController {
+    host: ReactiveControllerHost;
+    dropDownEl: HTMLElement;
+
+    constructor(host: ReactiveControllerHost, dropDownEl: HTMLElement) {
+        (this.host = host).addController(this);
+        this.dropDownEl = dropDownEl;
+    }
+
+    clickOutside(event: Event) {
+        const withinBoundaries = event.composedPath().includes(this.dropDownEl);
+        if (!withinBoundaries && this.dropDownEl && this.dropDownEl.classList.contains('open')) {
+            this.dropDownEl.classList.remove('open');
+        }
+    }
+
+    hostConnected() {
+        window.addEventListener('click', this.clickOutside.bind(this));
+    }
+
+    hostDisconnected() {
+        window.removeEventListener('click', this.clickOutside.bind(this));
+    }
+}
+
+export const ToniqDropdown = defineToniqElement({
+    tagName: 'toniq-dropdown',
+    props: {
+        list: [] as undefined | ToniqDropdownSelect[],
+        select: {} as undefined | ToniqDropdownSelect,
```

Comment:
default this to undefined
```suggestion
        select: undefined as undefined | ToniqDropdownSelect,
```